toqito/matrices/cyclic_permutation.py

```diff
@@ -0,0 +1,36 @@
+"""Generate a cyclic permutation matrix."""
+
+import numpy as np
+
+
+def cyclic_permutation_matrix(n: int) -> np.ndarray:
+    r"""Create the cyclic permutation matrix P for a given dimension n.
+
+    This function creates a cyclic permutation matrix which is a special type of square matrix
+    that represents a cyclic permutation of its rows.
+
+    Examples
+    ==========
+    Generate various cyclic permutation matrices.
+
+    >>> from this_module import cyclic_permutation_matrix
+    >>> n = 4
+    >>> cyclic_permutation_matrix = cyclic_permutation_matrix(n)
+    >>> print(cyclic_permutation_matrix)
+    [[0 0 0 1]
+    [1 0 0 0]
+    [0 1 0 0]
+    [0 0 1 0]]
```

Comment:
use `...` here for doctest to show a continued line on the next page. We will have a `doctest` workflow in the future to check the output in docstrings. 

I think the output is also supposed to be a numpy array, right? Specify this in the output.